### PR TITLE
fix(pina): preflight fallible account mutations

### PR DESCRIPTION
**Pull request by ifiokjr**
## Summary

- preflight active mutable borrows before reallocating or closing accounts
- reject single-call account growth above the Solana runtime limit before moving rent
- document the remaining cumulative-growth atomicity requirement
- add normal and Miri-oriented regressions for duplicate views and oversized growth

## Validation

- `cargo test --locked -p pina --all-features`
- `cargo check --locked -p pina --no-default-features --features account-resize`
- focused Miri suite previously passed (7/7)
- `dprint check` and `git diff --check`

Created on behalf of Ifiok Jr. (`@ifiokjr`), using OpenAI GPT-5.6 with high reasoning.

### CODE REVIEW

**Comment by ifiokjr** [comment]
@coderabbitai review

Created on behalf of Ifiok Jr. (@ifiokjr), using OpenAI GPT-5.6 with high reasoning.

File: crates/pina/src/cpi.rs [hunk 1]
```diff
@@ -455,6 +468,14 @@ fn realloc_account_inner(
 		return Ok(());
 	}
 
+	// `resize` would reject either condition after rent movement. The original
+	// serialized length is not exposed, so cumulative growth is still checked by
+	// the runtime and its error must be propagated by callers.
+	account.check_borrow_mut()?;
+	if new_size.saturating_sub(current_size) > MAX_PERMITTED_DATA_INCREASE {
```
Comment (ifiokjr) [inline crates/pina/src/cpi.rs:475]: The `realloc_rejects_oversized_single_growth_before_moving_rent` test covers `+ (MAX + 1)`. Add the equal-to-limit boundary case (`+ MAX_PERMITTED_DATA_INCREASE` exactly) asserting it proceeds past the preflight and reaches the rent path (success or the expected downstream behavior), to lock in the off-by-one and prevent a future `>=` regression from silently rejecting valid growth.

**Review by ifiokjr — COMMENTED** [review]
Strong, atomicity-preserving change. Preflighting the active mutable borrow and the single-growth limit *before* moving rent lamports is exactly right, and the honest doc note about the remaining cumulative-growth gap (Pinocchio not exposing the original serialized length) sets the right expectation. Sourcing the constant from pinocchio is a nice touch. One boundary-test suggestion.
